### PR TITLE
Add flag on search

### PR DESCRIPTION
**Pull request by guswnsxodlf**
- Add tests for search_flag
- Move PostManger and PostQuerySet to `board/models.py`, Add search flag option
- Modify 'post_list' view, template, scc due to addition of search flag
- Synchronizing search flag with template will be pull-requested later

### CODE REVIEW

File: kboard/board/views.py [hunk 1]
```diff
@@ -23,9 +23,14 @@ def new_post(request, board_slug):
...
-        posts = Post.objects.get_from_board(board).remain().search(query).order_by('-id')
+    search_info = {
+        'query': request.GET.get('query', ''),
+        'flag': request.GET.get('search_flag', '')
+    }
+
+    if search_info['query']:
+        posts = Post.objects.get_from_board(board).remain().search(search_info['flag'], search_info['query'])\
+            .order_by('-id')
     else:
```
Comment (cjh5414) [inline kboard/board/views.py:34]: if 'query' is '',  is search_info['query'] false?

**Review by cjh5414 — COMMENTED** [review]
LGTM

File: kboard/board/views.py (hunk 1 above)
Comment (guswnsxodlf) [reply kboard/board/views.py:34]: Yes :)
https://docs.python.org/3/library/stdtypes.html#truth-value-testing

**Comment by hyesun03** [comment]
LGTM

**Pull request merged**